x-pack/legacy/plugins/task_manager/task_store.test.ts

```diff
@@ -750,3 +987,28 @@ function generateFakeTasks(count: number = 1) {
     sort: ['a', _.random(1, 5)],
   }));
 }
+
+function concreteTaskInstanceAsSavedObject(
+  id: string,
+  type: string,
+  taskInstance: ConcreteTaskInstance
+): SavedObject {
+  return {
+    id,
+    type: 'task',
+    references: [],
+    version: taskInstance.version,
+    attributes: {
+      ...taskInstance,
+      ..._.mapValues(
+        _.pick(taskInstance, ['scheduledAt', 'startedAt', 'retryAt', 'runAt']),
+        (value: Date) => {
+          return value ? value.toISOString() : value;
+        }
+      ),
+      ..._.mapValues(_.pick(taskInstance, ['state', 'params']), (data: Record<string, any>) => {
```

Comment:
Yeah, I considered that, but thought we probably just want serialise all objects and dates.
Is that not the case?

Reply:
no harm in converting the dates to strings, but I think unneeded, but again (per the object serialization), perhaps I'm making assumptions about SO that aren't true :-) .